images/node-placeholder-scaler/scaler/calendar.py

```diff
@@ -83,31 +87,32 @@ def get_events(url: str):
 
     Mostly to deal with weird issues around url parsing and timezones
     """
-
     if url.startswith("file://"):
         path = url.split("://", 1)[1]
         with open(path) as f:
-            content = f.read()
+            calendar = IcsCalendarStream.calendar_from_ics(f.read())
     elif "://" not in url:
         path = url
         with open(path) as f:
-            content = f.read()
+            calendar = IcsCalendarStream.calendar_from_ics(f.read())
     else:
         r = requests.get(url)
         r.raise_for_status()
-        content = r.text
-
-    cal_tz = _get_cal_tz(content)
-
-    now = utcnow()
-    events = icalevents.parse_events(content, start=now, end=now)
-    for event in events:
-        # fix timezone for all-day events
-        if not event.start.tzinfo:
-            log.info(
-                f"Using calendar default timezone {cal_tz} for {_event_repr(event)}"
-            )
-            event.start = event.start.replace(tzinfo=cal_tz)
-        if not event.end.tzinfo:
-            event.end = event.end.replace(tzinfo=cal_tz)
+        calendar = IcsCalendarStream.calendar_from_ics(r.text)
+
+    cal_tz = _get_cal_tz(calendar)
+    #print(cal_tz)
+    #now = utcnow()
+    now = datetime.datetime(2023, 3, 2, 20, 0, 0, 0, tzinfo=cal_tz)
+    #events = icalevents.parse_events(content, start=now, end=now)
+    events = calendar.timeline.at_instant(now)
+    # for event in events:
+        # # fix timezone for all-day events
```

Comment:
sadly, all-day events don't have an event TZ, and the Event class value for `dtstart` and `dtend` is set at creation to either `datetime.datetime` or `datetime.time`.

in this case, it's `datetime.time` for all day events, and the builtins keep you from reassigning to a different (in this case, `datetime.datetime`) type.

boo.

```
>>> z[0].dtend = datetime.datetime(2023, 3, 2, 0, 0, tzinfo=zoneinfo.ZoneInfo(key='America/Los_Angeles'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pydantic/main.py", line 398, in pydantic.main.BaseModel.__setattr__
pydantic.error_wrappers.ValidationError: 1 validation error for Event
__root__
  Unexpected dtstart value '2023-03-02' was date but dtend value '2023-03-02 00:00:00-08:00' was datetime (type=value_error)
```